cmd/devp2p/ethcmd.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/cmd/devp2p/internal/ethtest"
+	"github.com/urfave/cli/v2"
+)
+
+var (
+	ethCommand = &cli.Command{
+		Name:  "eth",
+		Usage: "Eth Commands",
+		Subcommands: []*cli.Command{
+			&ethNewBlockCommand,
+		},
+		Flags: []cli.Flag{
+			chainFile,
+			genesisFile,
+		},
+	}
+	// Flags
+	chainFile = &cli.StringFlag{
+		Name:  "chain",
+		Usage: "Path to chain.rlp file.",
+		Value: "chain.rlp",
+	}
+	genesisFile = &cli.StringFlag{
+		Name:  "genesis",
+		Usage: "Path to genesis file.",
+		Value: "genesis.json",
+	}
+	// Subcommands
+	ethNewBlockCommand = cli.Command{
+		Name:  "new-msg",
+		Usage: "<node> <code> <msg>",
+		Description: `Example codes:
+  - GetBlockBodies: 0x15
+  - BlockBodies: 0x16
+  - NewBlock: 0x17
+  - NewPooledTransactionHashes: 0x18
+`,
+		Action: ethMessage,
+	}
+)
+
+// ethMessage peers with node, sends the provided message, then disconnects from the peer.
+func ethMessage(ctx *cli.Context) error {
+	if ctx.NArg() < 3 {
+		return errors.New("Three args needed: <node> <code> <msg>")
+	}
+	msgCode, ok := big.NewInt(0).SetString(ctx.Args().Get(1), 0)
+	if !ok {
+		return fmt.Errorf("unable to decode msg code")
+	}
+	code := msgCode.Uint64()
+	msg, err := hex.DecodeString(ctx.Args().Get(2))
+	if err != nil {
+		return fmt.Errorf("unable to decode msg: %s", err)
+	}
+	chain, err := ethtest.LoadChain(ctx.String(chainFile.Name), ctx.String(genesisFile.Name))
+	if err != nil {
+		return fmt.Errorf("error loading chain: %s", err)
+	}
+	conn, err := ethtest.Dial(getNodeArg(ctx))
+	if err != nil {
+		return fmt.Errorf("error dialing peer: %s", err)
+	}
+	// Peer with node.
+	status := &ethtest.Status{
+		ProtocolVersion: 66,
```

Comment:
This hard-coded version is going to be a problem. I think it should be passed as a flag.